### DISCUSSION TITLE
kiali1.33: Multicluster service health response include other nodes.

### DISCUSSION THREAD

**Discussion by LeslieKuo** [kiali/kiali#4055] [Multicluster] [answered]
In my view, the Multicluster `graph` request returns the homecluster and the related nodes in other clusters. The health requests consist of workload health and service health.  Workload requests return the workload health in **homecluster**,didn't include the related nodes in other clusters. I found the service request return the service health from homecluster and other cluster.
To illustrate clearly, I give a example. Entire service mesh consist of two cluster: cluster1 and cluster2. Take cluster1 as homecluster.
`graph ` requests will fetch all the details in cluster1,and the related part in cluster2.  ( graph: cluster1entire + cluster2part)
`health/workload` requests will fetch all the details in cluster1. (workload health: cluster1entire)
`health/service` request fetch all the details in cluster1 and cluster2.  (service health: cluster1entire+cluster2entire)

Graph decide the display in graph. So we will see cluster1entire + cluster2part. However, we only know the cluster1 workload health. So the related workload in cluster2 do not have health status?
In this picture the `reviews -v2 workload` in `tzotz cluster` workload health is unknown?
https://kiali.io/documentation/latest/features/#_graph_view_cluster_and_namespace_boxing 

One more question, I didn't figure out how `health/service` fetch all the mesh clusters service health at one time.
Thanks a lot.

**Comment by jshaughn** [comment, marked as answer]
Kiali is currently limited to a single Kubernetes client, communicating with what we currently call the **home cluster**.  It also communicates to a single Prometheus for metrics.  The graph will display traffic within the requested namespaces, as well as traffic inbound from, and outbound to, other namespaces.   It is currently assumed that only the home cluster will be reporting metrics to the Prometheus instance.   Although not a strict requirement, we have not tested unified metrics from multiple clusters.   Workload Health can only be performed for workloads on the home cluster as Kiali can query only the home kubernetes.  Traffic health is limited to what is in Prometheus, and so we can report on the service traffic, which can include the inbound and outbound edge traffic to other clusters, but only as reported by the proxies on the home cluster workloads.

How to better support of more complex environments is currently an ongoing discussion with the community, and we will be formalizing a proposal in the near term.

**Reply by LeslieKuo** [reply]
Big thanks!
